test/test_dir.rs

```diff
@@ -57,9 +57,3 @@ fn rewind() {
     assert_eq!(entries1, entries2);
     assert_eq!(entries2, entries3);
 }
-
-#[cfg(not(target_os = "haiku"))]
-#[test]
-fn ebadf() {
```

Comment:
This test is removed because this should not happen after the adoption of I/O safety.